ls1intum/Artemis
# Make uploads directory configurable.

### Checklist

### Motivation and Context
In a distributed setup, we want to configure the path where file uploads are stored, so that they can be made accessible from several (virtual) machines.

### Description
I've added a FilePathService that provides values for file paths there were previously constants where the path can be configured using `artemis.file-upload-path`.
As we cannot inject values into static fields, we have to use a workaround: Create an instance of the service that sets the static values.
We cannot use a non-static method as we use the file path in some of the domain classes, which should not have dependencies to services. In the future, we should probably redesign the classes in the domain package, to not depend on services. 

### Steps for Testing
1. Start Artemis without any config changes, upload e.g. a course icon and verify that it is saved in the `uploads` directory.
2. Change the value `artemis.file-upload-path` to a different folder and restart Artemis.
3. Upload a new file (e.g. again a course icon) and verify that it is stored in the newly configured folder.
